src/main/java/de/tum/in/www1/artemis/domain/hestia/CoverageFileReport.java

```diff
@@ -0,0 +1,77 @@
+package de.tum.in.www1.artemis.domain.hestia;
+
+import java.util.Set;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+
+@Entity
+@Table(name = "coverage_file_report")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class CoverageFileReport extends DomainObject {
+
+    @ManyToOne
+    @JsonIgnoreProperties("fileReports")
+    private CoverageReport fullReport;
+
+    @Column(name = "file_path")
+    private String filePath;
+
+    @Column(name = "line_count")
+    private Integer lineCount;
+
+    @Column(name = "covered_line_count")
+    private Integer coveredLineCount;
+
+    @OneToMany(mappedBy = "fileReport", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
```

Comment:
be careful with EAGER here. If you use it, then please document why it is necessary and intended.